### PR TITLE
feat(iOS): `signInWithBrowser()`

### PR DESCRIPTION
**Pull request by gao-sun**
## Summary

init iOS implementation of `signInWithBrowser()` using [Authentication Services](https://developer.apple.com/documentation/authenticationservices/authenticating_a_user_through_a_web_service)

To-dos:

- error handling
- fetch token by code
- persist storage

## Testing

in `SwiftUI Demo` app, signed in with logto.dev and returned desired URL.

image

## Reviewers

@logto-io/eng

### CODE REVIEW

File: Sources/LogtoClient/LogtoClient/LogtoClient+SignIn.swift [hunk 1]
```diff
@@ -32,9 +33,43 @@ public extension LogtoClient {
...
+            let codeChallenge = LogtoUtilities.generateCodeChallenge(codeVerifier: codeVerifier)
+
+            guard let authUri = try? LogtoCore.generateSignInUri(
+                authorizationEndpoint: oidcConfig.authorizationEndpoint,
+                clientId: logtoConfig.clientId,
+                redirectUri: "\(scheme)://callback",
+                codeChallenge: codeChallenge,
+                state: state,
+                scope: nil,
+                resource: nil
```
Comment (demonzoo) [inline Sources/LogtoClient/LogtoClient/LogtoClient+SignIn.swift:56]: I think we just aligned to use `scopes` and `resources`

File: Sources/LogtoClient/LogtoClient/LogtoClient+SignIn.swift [hunk 2]
```diff
@@ -32,9 +33,43 @@ public extension LogtoClient {
...
+    // TO-DO: implement full functions
     func signInWithBrowser() {
-        fetchOidcConfig { _, _ in
-            // To implement
+        fetchOidcConfig { [self] oidcConfig, _ in
+            guard let oidcConfig = oidcConfig else {
+                return
+            }
+
+            let scheme = "io.logto.SwiftUI-Demo"
```
Comment (xiaoyijun) [inline Sources/LogtoClient/LogtoClient/LogtoClient+SignIn.swift:44]: Maybe we should provide an approach for users to set up this `scheme`.

**Review by xiaoyijun — CHANGES_REQUESTED** [review]

**Review by simeng-li — APPROVED** [review]
LGTM

File: Sources/LogtoClient/LogtoClient/LogtoClient+SignIn.swift (hunk 2 above)
Comment (gao-sun) [reply Sources/LogtoClient/LogtoClient/LogtoClient+SignIn.swift:44]: yes, will do in the next pr

**Review by xiaoyijun — CHANGES_REQUESTED** [review]

**Review by xiaoyijun — APPROVED** [review]

File: Sources/LogtoClient/LogtoClient/LogtoClient+SignIn.swift (hunk 1 above)
Comment (gao-sun) [reply Sources/LogtoClient/LogtoClient/LogtoClient+SignIn.swift:56]: yes, will do in a separate PR

File: Sources/LogtoClient/LogtoClient/LogtoClient+SignIn.swift [hunk 3]
```diff
@@ -32,9 +33,43 @@ public extension LogtoClient {
...
+            let state = LogtoUtilities.generateState()
+            let codeVerifier = LogtoUtilities.generateCodeVerifier()
+            let codeChallenge = LogtoUtilities.generateCodeChallenge(codeVerifier: codeVerifier)
+
+            guard let authUri = try? LogtoCore.generateSignInUri(
+                authorizationEndpoint: oidcConfig.authorizationEndpoint,
+                clientId: logtoConfig.clientId,
+                redirectUri: "\(scheme)://callback",
```
[the remark this answers is not in the document]
Comment (gao-sun) [reply Sources/LogtoClient/LogtoClient/LogtoClient+SignIn.swift:52]: will do in the next pr

**Pull request merged**